### PR TITLE
daemon: Drop unused `oldConfig` parameter to update function

### PR DESCRIPTION
**Pull request by cgwalters**
Since nothing used it.  Also add a local variable for the target
URL to reduce noise.

### CODE REVIEW

**Comment by kikisdeliveryservice** [comment]
Looks like unit tests need to be updated for new arguments.

**Comment by ashcrow** [comment]
Looks like unit testing is happy now! :smile:

File: pkg/daemon/update_test.go [hunk 1]
```diff
@@ -49,15 +49,15 @@ func TestUpdateOS(t *testing.T) {
 	}
 
 	// Test when the machine configs match. No operation should occur
-	if err := d.updateOS(mcfg, mcfg); err != nil {
+	if err := d.updateOS(mcfg); err != nil {
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply pkg/daemon/update_test.go]: OK done :arrow_down:

**Comment by ashcrow** [comment]
Looks like flakes from the log, though it does also say that it's still running :man_shrugging: :smile:.

/test e2e-aws

**Pull request merged**